internal/plugincommon/discovery/processor.go

```diff
@@ -182,28 +180,55 @@ func aggregateObservations(
 	return obs
 }
 
-// Outcome comes to consensus on the contract addresses and updates the chainreader. It doesn't actually
-// return an Outcome.
+func getRouterConsensus(
+	lggr logger.Logger,
+	routerAddrs map[cciptypes.ChainSelector][][]byte,
+	thresh consensus.MultiThreshold[cciptypes.ChainSelector],
+) ([]byte, error) {
+	routerConsensus := consensus.GetConsensusMap(
+		lggr,
+		"router",
+		routerAddrs,
+		thresh,
+	)
+	// All the routers in routerAddrs are "valid", but we can only choose one.
+	routerCounts := make(map[string]int)
+	for _, addr := range routerConsensus {
+		routerCounts[string(addr)]++
+	}
```

Comment:
Would be useful to add this as a test case, but I'm trying to figure out if this is needed if you have 2F + 1 thresholds. If you have N = 3F + 1 and you do get consensus on a router address then you can just use that right away, because 2F+1 means that there's only F that observed otherwise, not enough to form another majority.

Reply:
Imagine srcChain1 has 2F+1 agreement on `0xRouter1` and srcChain2 has 2F+1 on `0xRouter2`. Which should we use?

Reply:
So each source chain will have a different router. If srcChain1 has 2FChain+1 honest majority on 0xRouter1 and srcChain2 has 2FChain+1 honest majority on 0xRouter2, we should just use each of these routers respectively for each chain. Isn't that what we're trying to get to? I don't think there's an ask to get the most common address across the sources, the router addresses will be different for different chains (in order to avoid confusion mostly)

Reply:
This was the approach @asoliman92 suggested. Your suggestion involves configuring multiple router addresses on the destination, how would you choose which one to use later? At the moment we're not configuring multiple bindings for any of the other contracts.

Reply:
Ah wait I think I'm confusing something - the routers in the "per source" config here refer to local router addresses, i.e those on the dest chain itself, so my above comment is incorrect.

Reply:
I think thats whats causing some confusion right now, maybe some renames would clarify it a bit more.